jax/_src/pallas/primitives.py

```diff
@@ -294,6 +294,43 @@ def _load_jvp(primals, tangents, args_tree, **params):
 
 ad.primitive_jvps[load_p] = _load_jvp
 
+def _pad_values_to_avoid_dynamic_slice_oob_shift(value,
+                                   slice_sizes, unpad=False):
+  """
+  DynamicSlice and DynamicUpdateSlice adjust the start index in cases where the
+  requested slice overruns the bounds of the array. This pads the array with
+  uninitialised values such that the requested slice will never overrun.
+
+  do not support out-of-bounds slices. Note the potentially surprising behavior
+  for the case where the requested slice overruns the bounds of the array; in
+  this case the start index is adjusted to return a slice of the requested size:
+
+  For example, if arr is [1.,2.,3.,4.] and a slice of size 4, start index 2 is
+  requested then the result will be [3.,4.,NaN,NaN] after padding, rather than
+  [1.,2.,3.,4.] from the unpadded array
+
+  unpad=True performs the inverse operation
+  """
+
+  padding_config = tuple((0, slice_size, 0) for slice_size in slice_sizes)
+  if (unpad):
+    padding_config = tuple((-low, -high, -interior)
+                           for (low, high, interior) in padding_config)
+  padding_value = _uninitialized_value(shape=(), dtype=value.dtype)
+  value = lax.pad(value,
+                  padding_config=padding_config,
+                  padding_value=padding_value)
+  return value
+
+_unpad_values_to_avoid_dynamic_slice_oob_shift = partial(
+  _pad_values_to_avoid_dynamic_slice_oob_shift, unpad=True)
+
+def _uninitialized_value(shape, dtype):
```

Comment:
This is a duplicate of https://github.com/google/jax/blob/1e48adc6982c2c774daf349d6d6574a1d041d772/jax/_src/pallas/pallas_call.py#L112-L117
is this okay, what do you suggest?

Reply:
I think it's fine to make this a public function here and import this function from pallas_call.

Reply:
Done (have also added behaviour for bool, where it counts False as uninitialised.)